fix: cohorts api permissions

dyudyunov opened the pull request:
## Description

We've discovered a bug while using the Gradebook MFE - users with course staff/instructor roles can create and manage cohorts in the course's Instructor tab BUT they can't use the cohorts filter and the API request returns 403
<img width="835" alt="Screenshot 2024-03-20 at 15 31 07" src="https://github.com/openedx/edx-platform/assets/47273130/87e64209-b4f3-413a-9a04-90885110e1a8">

The issue isn't reproducible for the Global Staff role's users as well as for **users with Forum Moderator roles assigned**.

## Decision
Considering that the course staff/instructor has most of the permissions for the course data manipulation I decided to:
- Fix lack of the permissions for course staff/instructor roles. 
- Allow course staff/admin users to use cohorts API.

## Steps To Reproduce
- create a course
- assign any user without Global Staff or Superuser permissions a course staff or course instructor (aka course admin) role
- navigate to the course's Instructor tab > cohorts > enable and create any number of cohorts for the course. You can use both Global Staff or course staff user.
- switch to the course staff/instructor user
- navigate to the course's Instructor tab > Student Admin > click View Gradebook
- inspect the browser console and try to use the Cohorts filter

## Expected Result
Course Staff/Instructor users can use the Cohorts filter on the Gradebook MFE and there is no 403 error for the `.../cohorts/` API call in the browser console.
![image](https://github.com/openedx/edx-platform/assets/47273130/933876e1-4c43-4cdd-998e-c8c10798429d)

## Notes
- The Discussions MFE behavior was not affected. 
Interestingly, only those users who have the Forum Moderator roles assigned can create discussion posts separated by cohorts. 
Even the Global Staff has no such option!
Anyway, it is out of the scope and I left it as it is.
<img width="1207" alt="Screenshot 2024-03-20 at 15 50 55" src="https://github.com/openedx/edx-platform/assets/47273130/36eee0da-1c73-40af-846b-a8dd1e5390c9">

openedx-webhooks commented:
Thanks for the pull request, @dyudyunov!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

dyudyunov committed:
fix: cohorts api permissions

Fix lack of the permissions for course staff/instructor roles.
Allows course staff/admin users use cohorts API, e.g. for
cohort filters in the Gradebook MFE.

dyudyunov commented:
@mphilbrick211 Hi

This one is ready for review 🙂

mphilbrick211 commented:
> @mphilbrick211 Hi
> 
> This one is ready for review 🙂

Thanks, @dyudyunov! I'm looking into getting a reviewer for this (and the backport).

dyudyunov commented:
@mphilbrick211 hi

It would be great to have this one reviewed so we could merge it along with the backport to Quince and include the fix in quince.3 release

dyudyunov commented:
@mphilbrick211 hi

Is there anyone able to review this?

mphilbrick211 commented:
@dyudyunov apologies for the delay. I've put this out to our Core Contributors for review.

sarina commented:
From a product perspective, this makes sense, but I'm not qualified to review the code. I'm hoping someone can pick up this review and merge it, given that product-wise this is good to go.

feanil reviewed (approved):
Code makes sense to me.

feanil commented:
@dyudyunov can you rebase this so I can merged this?

dyudyunov commented:
> @dyudyunov can you rebase this so I can merged this?

Hi Feanil
I rebased the branch
